packages/@aws-cdk/aws-rds/lib/props.ts

```diff
@@ -36,6 +36,29 @@ export class DatabaseClusterEngine {
     this.singleUserRotationApplication = singleUserRotationApplication;
     this.multiUserRotationApplication = multiUserRotationApplication;
   }
+
+  /**
+   * Get this engine's default parameter group family for given version
+   */
+  public getClusterParameterGroupFamily(engineVersion?: string): string {
+    if (this === DatabaseClusterEngine.AURORA) {
+      return 'aurora5.6';
+    } else if (this === DatabaseClusterEngine.AURORA_MYSQL) {
+      return 'aurora-mysql5.7';
+    } else if (this === DatabaseClusterEngine.AURORA_POSTGRESQL) {
+      if (engineVersion) {
+        if (engineVersion.startsWith('9')) {
+          return "aurora-postgresql9.6";
+        } else if (engineVersion.startsWith('10')) {
+          return "aurora-postgresql10";
+        } else if (engineVersion.startsWith('11')) {
+          return "aurora-postgresql11";
+        }
+      }
+      return "aurora-postgresql11";
+    }
+    throw new Error(`Unknown database engine: ${this.name}`);
+  }
```

Comment:
Sorry to keep pressing on this, but the implementation needs to be overhauled.

The constructor of `DatabaseClusterEngine` needs to be modified to take in the list of parameter group families which is then stored as an instance property. This method itself then uses the optional `engineVersion` parameter to determine which of the parameter group family to be returned.

Remember to add a unit test.

Reply:
Don't worry, I understand. I have changed it as per your suggestion and I'd say it is indeed better, but still not great. It's a necessary evil I think. Best thing would be if the CFN team just makes the Family property optional to let RDS figure out the proper family. Perhaps in the future and then we can clean this up.